### ISSUE TITLE
Console errors when using Mapbox GL 3 (Babel related)

### ISSUE THREAD

**Issue by matbmapspeople** [mapbox/mapbox-gl-js#12907]
**mapbox-gl-js version**: v3.0.0-beta.3

**browser**: Chrome, Safari, Firefox

### Steps to Trigger Behavior

I tried to run Mapbox GL beta3 in my web app.
 App uses:
  Angular CLI: 13.3.11
  Node: 16.13.2
  Package Manager: npm 8.1.2
  Angular: 13.3.6

 webpack.config.js includes:
```
 use: {
                    loader: 'babel-loader',
                    options: {
                        presets: ['@babel/preset-env']
                    }
         }
```

### Expected Behavior
No errors, running map.

### Actual Behavior

I followed migration guide to run the latest release of the new Mapbox GL 3 (beta3). I encountered some console errors that were related to the code that I am using. I commented my code out to make the map appear. Unfortunately, there are two console errors that for some reason I am not able to solve:

<img width="498" alt="Screenshot 2023-09-26 at 07 53 37" src="https://github.com/mapbox/mapbox-gl-js/assets/56337591/362bf56d-0f67-4ec8-956b-a87626d84b40">

Before I was using Mapbox GL 2 and everything worked in this case. Also I explicitly told the bundler to exclude node_modules when using Babel (webpack.config.js), but it did not work out.

Do you have any ideas or approach that might help me solving this problem?

**Comment by stepankuzmin** [comment]
Hi @matbmapspeople,

Could you please provide a minimal setup to help us address the issue?

**Comment by matbmapspeople** [comment]
Hey @stepankuzmin

Thank you for the reply.

I tried to create an online editor example, but I had some issues setting up proper dependencies. 
Although I encountered some problems there, I managed to create repository with exactly the same issue I am experiencing.

**Repo**: https://github.com/matbmapspeople/mapbox-gl-playground
The only thing that you have to do is: **npm install** and **npm start**. 

I hope that's acceptable and we can move on with it.
Let me know if you need anything else from me.

**Comment by tbHubs** [comment]
I am also experiencing this same issue with v3 within angular.

**Comment by stepankuzmin** [comment]
@matbmapspeople, it seems that you are still transpiling GL JS in your setup. At least `_asyncToGenerator` is not part of the GL JS, and it's probably Babel transpiling async functions to generators. Could you please check that you don't transpile GL JS?

**Comment by tbHubs** [comment]
@stepankuzmin If it works on the non-beta version, should we not assume the issue is with the new beta version and not the configuration of our transpiling handling?

**Comment by stepankuzmin** [comment]
It seems that the same error occurred in MapLibre https://github.com/maplibre/maplibre-gl-js/discussions/3210.

As far as I can see, Angular is forcing the transpilation to occur because `zone.js` is not compatible with native async/await:

https://github.com/angular/angular-cli/issues/25708
https://github.com/angular/angular/issues/31730
https://github.com/angular/angular-cli/pull/21552

We should look into different options to resolve this issue. /cc @mourner

**Comment by mourner** [comment]
@stepankuzmin short term the easiest is probably simply rewriting the `async / await` parts in the source that's included in the bundle to use old-school promises (there aren't that many occurences). It's a huge shame though, and long term we want to find a way that will allow us to use `async / await` in the bundle while not causing any issues.

**Comment by mourner** [comment]
We shipped a fix to this internally — will be a part of the final v3 release.

**Comment by matbmapspeople** [comment]
Thank you for the response and fix @mourner. Looking forward to final v3 release 👏

**Comment by stepankuzmin** [comment]
Hey @matbmapspeople 👋

Have you had a chance to upgrade to zoneless Angular? I know Angular 18+ added experimental zoneless support, and Angular 21 made it the default. As far as I know, this is the recommended approach now.

We're exploring native `async`/`await` in the bundle and want to understand how this affects folks:
- Are you still using Zone.js, or have you switched to zoneless?
- Would native `async`/`await` cause issues for your setup?

**Comment by matbmapspeople** [comment]
Thanks for checking in. This issue was on an Angular 13 app (Zone.js enabled, default Angular CLI setup). At the time I wasn’t using zoneless Angular—Angular 13 didn’t have it and I haven’t migrated this codebase yet.

So yes, the setup was Zone.js–based. If you’re moving toward native async/await in the bundle, my expectation is:
- In a Zone.js setup, native async/await shouldn’t break rendering, but it could change how patches/microtasks are tracked depending on how it’s bundled/transpiled.
- In a zoneless setup, it should be fine, but any app relying on Zone.js interception would obviously behave differently.

Happy to test again if you want a modern Angular 18+ zoneless reproduction.

**Comment by stepankuzmin** [comment]
Hey @matbmapspeople 👋

We've just released [v3.19.0-beta.1](https://github.com/mapbox/mapbox-gl-js/releases/tag/v3.19.0-beta.1) which includes native async/await in the bundle.

Would you be able to test it in your current setup and let us know if everything works?

**Comment by matbmapspeople** [comment]
Hey @stepankuzmin 
Seems like everything works as expected 👍 No errors in our setup.
